pkg/ifaceprobe/ifaceprobe.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+// Package ifaceprobe implements dynamic component of automatic kernel interface extraction.
+// Currently it discovers all /{dev,sys,proc} files, and collects coverage for open/read/write/mmap/ioctl
+// syscalls on these files. Later this allows to build file path <-> file_operations mapping.
+package ifaceprobe
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/flatrpc"
+	"github.com/google/syzkaller/pkg/fuzzer/queue"
+	"github.com/google/syzkaller/pkg/log"
+	"github.com/google/syzkaller/pkg/mgrconfig"
+	"github.com/google/syzkaller/pkg/symbolizer"
+	"github.com/google/syzkaller/prog"
+)
+
+// Info represents information about dynamically extracted information.
+type Info struct {
+	Files []FileInfo
+	PCs   []PCInfo
+}
+
+type FileInfo struct {
+	Name  string   // Full file name, e.g. /dev/null.
+	Cover []uint64 // Combined coverage for operations on the file.
+}
+
+type PCInfo struct {
+	PC   uint64
+	Func string
+	File string
+}
+
+func Globs() []string {
+	var globs []string
+	for _, path := range []string{"/dev", "/sys", "/proc"} {
+		// Our globs currently do not support recursion (#4906),
+		// so we append N "/*" parts manully. Some of the paths can be very deep, e.g. try:
+		// sudo find /sys -ls 2>/dev/null | sed "s#[^/]##g" | sort | uniq -c
+		for i := 1; i < 15; i++ {
+			globs = append(globs, path+strings.Repeat("/*", i))
+		}
+	}
+	return globs
+}
+
+func Run(ctx context.Context, cfg *mgrconfig.Config, exec queue.Executor, info *flatrpc.InfoRequest) (*Info, error) {
```

Comment:
It would help to have a comment on why we pass specifically a `*flatrpc.InfoRequest` object and what we expect to find/put there.